picker-starter/src/data/products.ts

```diff
@@ -641,3 +655,69 @@ export const products: MockProduct[] = tmpAssets
     } as MockProduct
   })
   .sort(compareName)
+
+export const getProductOptions: OptionsQuery = async () => {
+  const options: Option[] = categoryMockAssets.map<Option>((category) => {
+    return {
+      label: category.name,
+      value: category.name,
+    }
+  })
+
+  const response: MultiOption[] = [
+    {
+      type: 'single',
+      label: 'Category',
+      options: options,
+      defaultValue: undefined,
+      name: 'category',
+    },
+    {
+      type: 'multi',
+      options: options,
+      defaultValue: [],
+      label: 'Categories',
+      name: 'categoryMulti',
+    },
+  ]
+
+  return delayed(randomDelay(), response)
+}
+
+export const queryProducts: ItemQuery = async ({
+  search_term,
+  page,
+  per_page,
+  userOptions,
+}) => {
+  const matchCategory =
+    (categoryName: undefined | string) => (product: MockProduct) => {
+      if (typeof categoryName === 'undefined') {
+        return true
+      }
+      return product.category === categoryName
+    }
+  const matchCategories =
+    (categoryNames: string[]) => (product: MockProduct) => {
+      if (categoryNames.length === 0) {
+        return true
+      }
+      return categoryNames.some(
+        (categoryName) => product.category === categoryName,
+      )
+    }
+
+  const allSearchResults = products
+    .filter(matchCategory(userOptions['category'] as string | undefined))
+    .filter(matchCategories(userOptions['categoryMulti'] as string[]))
+    .filter(matchItem(search_term))
+
+  const paginatedResults = getPage(allSearchResults, page, per_page)
+  const response = {
+    items: paginatedResults,
+    pageInfo: {
+      totalCount: allSearchResults.length,
+    },
+  }
+  return delayed(randomDelay(), response)
```

Comment:
what is this delay?

Reply:
I think this is just to make the behaviour look like an actual call to an API.

Reply:
That's true, @BibiSebi. 

By simulating an actual API call, the `skeleton` has time to be visible before showing the real items.

![image](https://github.com/storyblok/field-type-examples/assets/1240591/dfa79674-a74f-4efe-a22e-e5f2667f96a0)

Reply:
ah okay! then.. do we want to keep it, or not? 🤔

Reply:
I'd keep it in this version and add a comment, something like:

```
// It mimics an API call by adding a delay before return the data.
return delayed(randomDelay(), response)
```

What do you think?